### ISSUE TITLE
Client Processes Token to Incorrect Identity

### ISSUE THREAD

**Issue by wkoutre** [twilio/TwilioChatJsReactNative#13]
### Example Users:

User A:
`userId`: 123

User B:
`userId`: 987

### Steps to reproduce:
Keep in mind, all this happens (and is related to) a single  "app session" -- that is, the app is _never_ force closed/quit between steps.

```
import { Client as TwilioChatClient } from "twilio-chat";
import { AccessManager as TwilioAccessManager } from "twilio-common";
```

- Login as User A
- Twilio token is generated with identity `123`
- Twilio Client is created with this token:

```MainTwilioClient = await TwilioChatClient.create(twilioToken, createTwilioClientOptions;```

- Access Manager is also created:

```this.accessManager = new TwilioAccessManager(twilioToken);```

- `client.user.state.identity` is `123`
- `accessManager.identity` is `123`
- User A can do everything as expected with Twilio
- User A logs out
- `client`'s listeners are all removed -- I do NOT call `client.shutdown()` for reasons to be explained later
- `client = null`
- Login as User B
- Twilio token is generated with identity `987`
- Twilio Client is created with this token
- Twilio Access Manager is created with this token
- `accessManager.identity` is `123`
- `client.user.state.identity` is still `123` when it **SHOULD BE** `987`

Note: `client.shutdown()` results in the inability to create ANY Client instances unless the app is reloaded.

Basically, I'm only able to create an instance of the Twilio Client with a single user per "app session".

If I logout with User A and back in with User B...

- the token is successfully created for User B
- `AccessManager` takes in this token and returns an instance of `AccessManager` associated with User B's `identity`
- `Client` takes User B's token and returns an instance with the `identity` of User A, which results in `403 - Forbidden` when calling something like `client.getUserChannelDescriptors()`

Reloading the app clears everything up, but that's not a viable solution for a production app.

I've replicated this with 100% consistency for hours, but if code samples are needed please let me know.

**Comment by aleksandrsivanovs** [comment]
Hi!

Our recommended way would be to shutdown the client and then create new client with new token.
And without shutdown of the first client it's not fully supported (and won't be fully supported) to create new client.

the core problem here is that shutdown doesn't work correctly (if you say that you can't create the second client after shutdown). I'll try to reproduce by myself, but, that would be awesome to have the clean debug logs with clearly reproduced problem, will you be able to provide those?

also, if you can - please do not use accessmanager for reproducing this issue - then we can exclude additional dependency.

Aleksandrs

P.S. I've seen something similar recently - the problem there was that customer was re-using the `clientoptions` object. Can you try to create the new client with new token and completely new client options? (i.e. do not reuse object from the first time). Also, code snippet of how exactly you do create -> dosmth -> shutdown -> create 2nd client -> dosmth would be awesome to have.

**Comment by wkoutre** [comment]
@aleksandrsivanovs Thanks for the quick reply.

Your P.S. note was the exact issue I was facing. I was re-using the `clientOptions` object.

Calling `shutdown()` and using a new client options object cleared this issue up for me. Definitely something easy to miss to add to the docs!

I'm considering this issue to be resolved. Please close as necessary. Thanks again!

**Comment by aleksandrsivanovs** [comment]
good to hear :)
this will be fixed in upcoming version.

Aleksandrs

**Comment by aleksandrsivanovs** [comment]
we have just released twilio-chat 2.2.0 @npmjs.
the sample is updated, now it should work reusing the same clientoptions: https://github.com/twilio/TwilioChatJsReactNative/commit/743a74165cfc20072edc15d4baa2a23b43414f28
